coderd/presets.go

```diff
@@ -64,13 +41,21 @@ func (api *API) templateVersionPresetParameters(rw http.ResponseWriter, r *http.
 		return
 	}
 
-	var res []codersdk.PresetParameter
-	for _, presetParam := range presetParams {
-		res = append(res, codersdk.PresetParameter{
-			PresetID: presetParam.TemplateVersionPresetID,
-			Name:     presetParam.Name,
-			Value:    presetParam.Value,
-		})
+	var res []codersdk.Preset
+	for _, preset := range presets {
+		sdkPreset := codersdk.Preset{
+			ID:   preset.ID,
+			Name: preset.Name,
+		}
+		for _, presetParam := range presetParams {
+			if presetParam.TemplateVersionPresetID == preset.ID {
```

Comment:
Isn't this check redundant? You're only fetching params for this template version in `GetPresetParametersByTemplateVersionID` right?

Reply:
good catch. removing.